### ISSUE TITLE
[c-sharp] Properties is missing ? (nullable) when graphtype is allowing null and the same for deprecated

### ISSUE THREAD

**Issue by madsrode** [dotansimha/graphql-code-generator#4214]
**Describe the bug**
When a field is marked as nullable and/or deprecated is it correctly generated in ex. typescript, but in C# is both missing from the generated file.

**To Reproduce**
Sandbox link: https://codesandbox.io/s/lingering-paper-0zysu

1. My GraphQL schema:

```graphql
type Query {
    user(id: ID!): User!
}

type User {
    id: ID!
    username: String!
    email: String!
    nullableFloat: Float @deprecated(reason: "Just gone.")
}

```

2. My GraphQL operations:

```graphql
query user {
    user(id: 1) {
        id
        username
        email
        nullableFloat
    }
}

```

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
documents: document.graphql
generates:
  types.ts:
    plugins:
      - typescript
  Gen.cs:
    plugins:
      - c-sharp

```

**Expected behavior**
Gen.cs file with User class should look like this: (with ? at the type of nullableFloat and Obsolete attribute above the field.

`  #region User
  public class User {
    #region members
      
            [JsonProperty("id")]
            public string id { get; set;}
          
    
            [JsonProperty("username")]
            public string username { get; set;}
          
    
            [JsonProperty("email")]
            public string email { get; set;}
          
    
            [JsonProperty("nullableFloat")]
            [Obsolete("Just gone.")]
            public float? nullableFloat { get; set;}
          
    #endregion
  }
  #endregion`

**Environment:**

- OS: 
- `@graphql-codegen/...`: 
- NodeJS:

**Comment by cklam2** [comment]
Indeed. Also probably due to the missing nullable marker I see quite some warnings in the generated types file. For example
```
if (this.hasPhone != null) {
   d["hasPhone"] = hasPhone;
}
```

Warning shown for this excerpt:
```
bool User.hasPhone { get; set; }
The result of the expression is always 'true' since a value of type 'bool' is never equal to 'null' of type 'bool?'
```

**Comment by cklam2** [comment]
Attempting to solve this + some other improvements:
- Nullable markers for fields that use value type and are optional (so no support yet for nullable reference types)
- Null check only for nullable value types and reference types
- Mark field obsolete when the @deprecated decorator found
- Comment header. Currently only for fields, hopefully can do it for all fields/classes, etc
- Boolean => bool

Just FYI and hopefully can do a pull request soon.

**Comment by dotansimha** [comment]
Fixed in v1.15.4
